### PR TITLE
JNG-4989 update jsl documentation

### PR DESCRIPTION
**Pull request by gprivi**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4989" title="JNG-4989" target="_blank">StoryJNG-4989</a>  Update JSL documentation
  </td></table>
  <br />
 

JNG-4989 update jsl documentation

### CODE REVIEW

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityMemberDeclaration.eol [hunk 1]
```diff
@@ -1,7 +1,7 @@
-@cached
-operation JSL!EntityMemberDeclaration getReferenceType() : JSL!SingleType {
-    return jslUtils.getReferenceType(self);
-}
+//@cached
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityMemberDeclaration.eol]: If its not used, just remove it.

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/transferRelationDeclaration.eol [hunk 2]
```diff
@@ -1,7 +1,7 @@
-@cached
-operation JSL!TransferRelationDeclaration getReferenceType() : JSL!TransferDeclaration {
-    return jslUtils.getReferenceType(self);
-}
+//@cached
+//operation JSL!TransferRelationDeclaration getReferenceType() : JSL!TransferDeclaration {
+//    return jslUtils.getReferenceType(self);
+//}
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/transferRelationDeclaration.eol]: Deas code, just remove it.

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/jslToPsm.etl [hunk 3]
```diff
@@ -30,6 +26,9 @@ import "modules/type/type.etl";
 
 pre {
     JUDOPSM.target.resource.contents.add(defaultModelName.equivalent("CreateRootModel"));
+
+	// sets the extended modifier properties of JSL model
+	// setModifiers();
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/jslToPsm.etl]: Dead code, remove it.

**Review by robertcsakany — CHANGES_REQUESTED** [review]

**Review by Madmat8 — APPROVED** [review]

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityMemberDeclaration.eol (hunk 1 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/entityMemberDeclaration.eol]: removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/jslToPsm.etl (hunk 3 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/jslToPsm.etl]: removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/transferRelationDeclaration.eol (hunk 2 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/transferRelationDeclaration.eol]: removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/errorDeclaration.eol [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
+//    }
+//
+//    if (self.`extends`.isDefined()) {
+//        for (f in self.`extends`.getInheritedFields()) {
+//            fields.add(f);
+//        }
+//    }
+//
+//    return fields;
+//}
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/errorDeclaration.eol]: Is there any intention to keep this? Any alternative for Error transformation?

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/errorField.eol [hunk 5]
```diff
@@ -1,4 +1,4 @@
-@cached
-operation JSL!ErrorField getId(): String {
-    return (self.eContainer.getId() + "/" + self.name);
-}
+//@cached
+//operation JSL!ErrorField getId(): String {
+//    return (self.eContainer.getId() + "/" + self.name);
+//}
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/errorField.eol]: Is therre any intention of the keep of this? Where is the alternative?

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/data/errorType.etl [hunk 6]
```diff
@@ -1,30 +1,30 @@
...
+//                    t.attributes.add(inheritedField.equivalentWithPrefix("CreateTransferAttributeForErrorTransferObjectType", t.getId()));
+//                } else {
+//                    throw "Unsupported Error Field Type: " + inheritedField.name;
+//                }
+//            }
+//        }
+//
+//        log.debug("Created ErrorType: " + t.name);
+//    }
+//
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/data/errorType.etl]: Is there any intention to keep? Where is the Error transfer generation?

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl [hunk 7]
```diff
@@ -98,48 +97,98 @@ rule CreateMappedTransferAttributeEntityDefault
...
+//    to t: JUDOPSM!TransferAttribute {
+//        guard: s.leftValue.declaration.getReferenceType().isKindOf(JSL!PrimitiveDeclaration) and s.leftValue.declaration.eContainer.map.isDefined()
+//        t.name = defaultDefaultNamePrefix + s.leftValue.declaration.name + defaultDefaultNameMidfix + s.leftValue.declaration.eContainer.name + defaultDefaultNamePostfix;
+//        t.binding = s.equivalent("CreateDefaultDataPropertyForMappedTransferObjectConstructor");	  
+//        t.dataType = s.leftValue.declaration.referenceType.getPSMEquivalent();
+//        t.setId("(jsl/" + s.leftValue.declaration.getId() + ")/CreateTransferEntityDefaultValueAttributeForMappedTransferObjectConstructor");
+//        s.eContainer.eContainer.getPSMEquivalent().attributes.add(t);
+//        log.debug("Created TransferAttribute DefaultValue (Field) for MappedTransferObjectConstructor: [" + t.name + "] into [" + t.eContainer.name + "]");
+//}
+
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl]: Please remove this

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl [hunk 8]
```diff
@@ -98,48 +97,98 @@ rule CreateMappedTransferAttributeEntityDefault
...
+//    to t: JUDOPSM!TransferAttribute {
+//        guard: s.leftValue.declaration.getReferenceType().isKindOf(JSL!PrimitiveDeclaration) and s.leftValue.declaration.eContainer.map.isUndefined()
+//        t.name = defaultDefaultNamePrefix + s.leftValue.declaration.name + defaultDefaultNameMidfix + s.leftValue.declaration.eContainer.name + defaultDefaultNamePostfix;
+//        t.binding = s.equivalent("CreateDefaultStaticDataForUnmappedTransferObjectConstructor");	  
+//        t.dataType = s.leftValue.declaration.referenceType.getPSMEquivalent();
+//        t.setId("(jsl/" + s.leftValue.declaration.getId() + ")/CreateTransferEntityDefaultValueAttributeForUnmappedTransferObjectConstructor");
+//        s.eContainer.eContainer.getPSMEquivalent().attributes.add(t);
+//        log.debug("Created TransferAttribute DefaultValue (Field) for UnmappedTransferObjectConstructor: [" + t.name + "] into [" + t.eContainer.name + "]");
+//}
+
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl]: Please remove this.

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl [hunk 9]
```diff
@@ -98,48 +97,98 @@ rule CreateMappedTransferAttributeEntityDefault
...
+//    transform s : JSL!TransferDataAssignment
+//    to t : JUDOPSM!Annotation {
+//        guard: s.leftValue.declaration.getReferenceType().isKindOf(JSL!PrimitiveDeclaration) and s.leftValue.declaration.eContainer.map.isDefined()
+//
+//        t.name = "TransferObjectAttributeWithDefaultValue";
+//        t.setId("(jsl/" + s.leftValue.declaration.getId() + ")/CreateAnnotationForTransferEntityDefaultValueAttributeForMappedTransferObjectConstructor");
+//        s.equivalent("CreateTransferEntityDefaultValueAttributeForMappedTransferObjectConstructor").annotations.add(t);
+//        log.debug("Created annotation for TransferAttribute DefaultValue (Field) for MappedTransferObjectConstructor: [" + t.name + "]");
+//}
+
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl]: Please remove this

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl [hunk 10]
```diff
@@ -98,48 +97,98 @@ rule CreateMappedTransferAttributeEntityDefault
...
+//rule CreateAnnotationForTransferEntityDefaultValueAttributeForUnmappedTransferObjectConstructor
+//    transform s : JSL!TransferDataAssignment
+//    to t : JUDOPSM!Annotation {
+//        guard: s.leftValue.declaration.getReferenceType().isKindOf(JSL!PrimitiveDeclaration) and s.leftValue.declaration.eContainer.map.isUndefined()
+//
+//        t.name = "TransferObjectAttributeWithDefaultValue";
+//        t.setId("(jsl/" + s.leftValue.declaration.getId() + ")/CreateAnnotationForTransferEntityDefaultValueAttributeForUnmappedTransferObjectConstructor");
+//        s.equivalent("CreateTransferEntityDefaultValueAttributeForUnmappedTransferObjectConstructor").annotations.add(t);
+//        log.debug("Created annotation for TransferAttribute DefaultValue (Field) for UnmappedTransferObjectConstructor: [" + t.name + "]");
+//}
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl]: Please remove this.

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferRelation.etl [hunk 11]
```diff
@@ -24,17 +24,17 @@ rule CreateTransientTransferObjectRelationForTransferRelationDeclaration
...
     extends AbstractCreateTransferObjectRelationForTransferRelationDeclaration {
-        guard: not s.maps and not s.reads and not s.eContainer.automap
+        guard: not s.maps() and not s.reads()
 
         t.setId("(jsl/" + s.getId() + ")/CreateTransientTransferObjectRelationForTransferRelationDeclaration");
         t.cardinality = s.equivalentWithPrefix("CreateCardinalityForTransferRelationDeclaration", t.getId());
 
-        var def = JSL!TransferDefault.all().selectOne(d | d.member.reference == s);
-        if (def.isDefined()) {
+        // var def = JSL!TransferDataAssignment.all().selectOne(d | d.leftValue.declaration == s);
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferRelation.etl]: Please remove this

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferRelation.etl [hunk 12]
```diff
@@ -86,9 +86,9 @@ rule CreateMappedTransferObjectEmbeddedRelationForTransferRelationDeclaration
       }
 
 
-      var def = JSL!TransferDefault.all().selectOne(d | d.member.reference == s);
-      if (def.isDefined()) {
-        t.defaultValue = def.equivalent("CreateDefaultNavigationPropertyForMappedTransferObjectConstructor");
+      // var def = JSL!TransferDataAssignment.all().selectOne(d | d.leftValue.declaration == s);
```
Comment (robertcsakany) [inline judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferRelation.etl]: Please remove this.

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/errorDeclaration.eol (hunk 4 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/errorDeclaration.eol]: file removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/data/errorType.etl (hunk 6 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/data/errorType.etl]: file removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/errorField.eol (hunk 5 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/operations/jsl/data/errorField.eol]: file removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl (hunk 7 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl]: removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl (hunk 8 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl]: removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl (hunk 9 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl]: removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl (hunk 10 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferAttribute.etl]: removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferRelation.etl (hunk 11 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferRelation.etl]: removed

File: judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferRelation.etl (hunk 12 above)
Comment (gprivi) [reply judo-tatami-jsl-jsl2psm/src/main/epsilon/transformations/psm/modules/structure/transferDeclarationTransferRelation.etl]: removed

**Pull request merged**